plugins/bcli.c

```diff
@@ -834,14 +837,62 @@ static void bitcoind_failure(struct plugin *p, const char *error_message)
 		      args_string(cmd, cmd));
 }
 
-static void wait_for_bitcoind(struct plugin *p)
+/* Do some sanity checks on bitcoind based on the output of `getnetworkinfo`. */
+static void parse_getnetworkinfo_result(struct plugin *p, const char *buf)
+{
+	const jsmntok_t *result, *versiontok, *relaytok;
+	bool valid, tx_relay;
+
+	result = json_parse_input(NULL,
+				  buf, strlen(buf),
+				  &valid);
+	if (!result || !valid)
+		plugin_err(p, "No or invalid response to '%s' ? Got '%s'. Can not"
+			      " continue without proceeding to sanity checks.",
+			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
+
+	/* Check that we have a fully-featured `estimatesmartfee`. */
+	versiontok = json_get_member(buf, result, "version");
+	if (!versiontok)
+		plugin_err(p, "No 'version' in '%s' ? Got '%s'. Can not"
+			      " continue without proceeding to sanity checks.",
+			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
+
+	if (!json_to_u32(buf, versiontok, &bitcoind->version))
+		plugin_err(p, "Invalid 'version' in '%s' ? Got '%s'. Can not"
+			      " continue without proceeding to sanity checks.",
+			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
+
+	if (bitcoind->version < 160000)
+		plugin_err(p, "Unsupported bitcoind version, you need to update"
+			      " Bitcoin Core.");
+
+	/* We don't support 'blocksonly', as we rely on transaction relay for fee
+	 * estimates. */
+	relaytok = json_get_member(buf, result, "localrelay");
+	if (!relaytok || !json_to_bool(buf, relaytok, &tx_relay))
+		plugin_err(p, "No 'localrelay' in '%s' ? Got '%s'. Can not"
+			      " continue without proceeding to sanity checks.",
+			      gather_args(bitcoind, "getnetworkinfo", NULL), buf);
+
+	if (!tx_relay)
+		plugin_err(p, "The 'blocksonly' mode of bitcoind, or any option "
+			      "deactivating transaction relay is not supported.");
```

Comment:
Maybe offer a suggestion of how to fix it? i.e. 

```suggestion
			      "deactivating transaction relay is not supported. Remove -blocksonly from your bitcoind config");
```

Reply:
Seems a bit redundant ?..
> The 'blocksonly' mode of bitcoind [...] is not supported. Remove -blocksonly from your bitcoind config

Reply:
Agree as well.